src/integration/integration_api_scalingengine_test.go

```diff
@@ -0,0 +1,206 @@
+package integration_test
+
+import (
+	"autoscaler/cf"
+	"autoscaler/models"
+	"encoding/json"
+	"fmt"
+	. "integration"
+	"net/http"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+type ScalingHistoryResult struct {
+	TotalResults int                        `json:"total_results"`
+	TotalPages   int                        `json:"total_pages"`
+	Page         int                        `json:"page"`
+	Resources    []models.AppScalingHistory `json:"resources"`
+}
+
+var _ = Describe("Integration_Api_ScalingEngine", func() {
+	var (
+		initInstanceCount int = 2
+		appId             string
+		parameters        map[string]string
+		history           *models.AppScalingHistory
+	)
+
+	BeforeEach(func() {
+		initializeHttpClient("api.crt", "api.key", "autoscaler-ca.crt", apiScalingEngineHttpRequestTimeout)
+		startFakeCCNOAAUAA(initInstanceCount)
+		scalingEngineConfPath = components.PrepareScalingEngineConfig(dbUrl, components.Ports[ScalingEngine], fakeCCNOAAUAA.URL(), cf.GrantTypePassword, tmpDir, consulRunner.ConsulCluster())
+		startScalingEngine()
+
+		apiServerConfPath = components.PrepareApiServerConfig(components.Ports[APIServer], dbUrl, fmt.Sprintf("https://127.0.0.1:%d", components.Ports[Scheduler]), fmt.Sprintf("https://127.0.0.1:%d", components.Ports[ScalingEngine]), tmpDir)
+		startApiServer()
+		appId = getRandomId()
+
+	})
+
+	AfterEach(func() {
+		stopApiServer()
+		stopScalingEngine()
+	})
+	Describe("Get scaling histories", func() {
+		Context("ScalingEngine is down", func() {
+			JustBeforeEach(func() {
+				stopScalingEngine()
+				parameters = map[string]string{"start-time": "1111", "end-time": "9999", "order": "desc", "page": "0", "results-per-page": "5"}
+			})
+
+			AfterEach(func() {
+				startScalingEngine()
```

Comment:
why we start again?  as long as you make sure stopScalingEngine() works when the scaling engine is stopped already.